compiler/rustc_trait_selection/src/traits/vtable.rs

```diff
@@ -232,6 +234,32 @@ fn own_existential_vtable_entries_iter(
     own_entries
 }
 
+fn trait_object_ty<'tcx>(tcx: TyCtxt<'tcx>, poly_trait_ref: ty::PolyTraitRef<'tcx>) -> Ty<'tcx> {
```

Comment:
Please assert that `!poly_trait_ref.has_non_region_param()`, since this only works if the trait ref is fully monomorphized.

Reply:
Done.